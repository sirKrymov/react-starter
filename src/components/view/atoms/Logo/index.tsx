// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement, memo } from 'react';

import { krymovLogo } from 'constants/images';

import './styles.scss';

interface IProps {
  id: string;
}

function LogoBase({ id }: IProps): ReactElement {
  return (
    <div className="logo" id={id}>
      <img className="img-contain" src={krymovLogo} alt="logo" />
    </div>
  );
}

export const Logo = memo(LogoBase);
