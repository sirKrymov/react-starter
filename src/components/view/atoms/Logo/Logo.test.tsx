import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Logo } from '../Logo';

describe('<Logo> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(<Logo id="main-logo" />);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
