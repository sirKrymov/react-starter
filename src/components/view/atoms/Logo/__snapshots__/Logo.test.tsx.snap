// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="logo"
  id="main-logo"
>
  <img
    alt="logo"
    className="img-contain"
    src="krymov-logo.png"
  />
</div>
`;
